### PR TITLE
Create `lib-subject-viewers` & make importable for `lib-classifier`

### PR DESCRIPTION
**Pull request by kieftrav**
## Package

`lib-subject-viewers`

## Linked Issue and/or Talk Post
n/a

## Describe your changes
- Create the base structure for the `lib-subject-viewers` package in FEM. 
- Make sure `lib-classifier` is able to import the package

## How to Review
1) Run `yarn panic && yarn bootstrap` to create a fresh codebase
1) Run all commands in the `package.json` to ensure they all succeed and none error out
1) Check that the ProtoViewer component renders properly in Storybook
1) Test adding the component to the `lib-classifier` package. Go into the `lib-classifier/dev/components/App.js` file and import the `ProtoViewer` at the top and then test that it renders properly in the component.

```js
import { ProtoViewer } from '@zooniverse/subject-viewers'
```

## General
- [ ] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [ ] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [ ] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

### CODE REVIEW

**Comment by kieftrav** [comment]
@goplayoutside3 - this is now ready for review

File: packages/lib-subject-viewers/README.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+- [React.js](https://reactjs.org/) - Component, virtual DOM based javascript library
+- [styled-components](https://www.styled-components.com/) - CSS in JS for styling and theming
+- [react-i18next](https://react.i18next.com/) - Internationalization framework for React.
+
+Testing is done by
+
+- [Mocha](https://mochajs.org/) - test runner
+- [Chai](https://www.chaijs.com/) - BDD/TDD assertion library
+- [Sinon](https://sinonjs.org) - test spies, mocks, and stubs
+- [Enzyme](https://airbnb.io/enzyme/) - testing utility for React
```
Comment (goplayoutside3) [inline packages/lib-subject-viewers/README.md:53]: Please revise this section. lib-subject-viewers does not use `styled-components` yet, and will never use `enzyme`. Translations packages should wait for actual content and a separate PR.

File: packages/lib-subject-viewers/.storybook/preview.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
+import { INITIAL_VIEWPORTS } from '@storybook/addon-viewport'
+import i18n from '../src/translations/i18n'
+import zooTheme from '@zooniverse/grommet-theme'
+import { Grommet } from 'grommet'
```
Comment (goplayoutside3) [inline packages/lib-subject-viewers/.storybook/preview.js:4]: These lines are copy-pasted from another library. Please refactor to match the needs of lib-subject-viewers. `i18n` should wait for actual content, and you're missing grommet + styling packages from a `peerDependencies` section in package.json.

File: packages/lib-subject-viewers/package.json [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    "html-webpack-plugin": "~5.6.0",
+    "mocha": "~10.2.0",
+    "sinon": "~17.0.0",
+    "sinon-chai": "~3.7.0",
+    "storybook": "~7.6.11",
+    "storybook-react-i18next": "~2.0.1",
+    "webpack": "~5.90.0",
+    "webpack-cli": "~5.1.0",
+    "webpack-dev-server": "~5.0.3"
+  },
```
Comment (goplayoutside3) [inline packages/lib-subject-viewers/package.json:61]: Missing a peerDependencies section here

File: packages/lib-subject-viewers/.babelrc [hunk 4]
```diff
@@ -0,0 +1,48 @@
+{
+  "plugins": [
+    "babel-plugin-styled-components",
+    [
+      "transform-imports",
+      {
+        "lodash": {
+          "transform": "lodash/${member}"
+        }
```
Comment (goplayoutside3) [inline packages/lib-subject-viewers/.babelrc:9]: Is lodash used in the viewer component? If not please remove. This was copy pasted from another library.

File: packages/lib-subject-viewers/.gitignore [hunk 5]
```diff
@@ -0,0 +1,5 @@
+build
+dist
+
+# Ignore static asset generation
+storybook-static/*
```
Comment (goplayoutside3) [inline packages/lib-subject-viewers/.gitignore:5]: Why is this included for this library?

File: packages/lib-subject-viewers/.npmignore [hunk 6]
```diff
@@ -0,0 +1,13 @@
+.cache
```
Comment (goplayoutside3) [inline packages/lib-subject-viewers/.npmignore:1]: This is copy pasted from lib-react-components and outdated. Please delete until ready to publish to npm, or revise specifically for this library.

File: packages/lib-subject-viewers/.storybook/main.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
+const config = {
+  stories: ["../src/**/*.stories.js"],
+  addons: [
+    "@storybook/addon-a11y",
+    "@storybook/addon-essentials",
+    "storybook-react-i18next",
```
Comment (goplayoutside3) [inline packages/lib-subject-viewers/.storybook/main.js:6]:
```suggestion
```

`i18n` should wait for actual content and a separate PR.

File: packages/lib-subject-viewers/.storybook/manager.js [hunk 8]
```diff
@@ -0,0 +1,12 @@
+import { addons } from "@storybook/addons";
+import { create } from "@storybook/theming/create";
+
+const theme = create({
+  base: "light",
+  brandTitle: "Zooniverse React Components",
```
Comment (goplayoutside3) [inline packages/lib-subject-viewers/.storybook/manager.js:6]: This is copy pasted from lib-react-components. Please retitle.

File: packages/lib-subject-viewers/preview.js [hunk 9]
```diff
@@ -0,0 +1,61 @@
+import { INITIAL_VIEWPORTS } from '@storybook/addon-viewport'
```
Comment (goplayoutside3) [inline packages/lib-subject-viewers/preview.js:1]: Is this a duplicate of `/.storybook/preview.js`?

File: packages/lib-subject-viewers/src/translations/i18n.js [hunk 10]
```diff
@@ -0,0 +1,28 @@
+import i18n from 'i18next'
```
Comment (goplayoutside3) [inline packages/lib-subject-viewers/src/translations/i18n.js:1]: Please remove all translation related files and content for a separate PR. Translations will be implemented only if there is platform-level labels specific to this viewer.

File: packages/lib-subject-viewers/test/setup.js [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+  disconnect () {}
+  observe () {}
+  unobserve () {}
+}
+
+class IntersectionObserver {
+  disconnect () {}
+  observe () {}
+  unobserve () {}
+}
```
Comment (goplayoutside3) [inline packages/lib-subject-viewers/test/setup.js:30]: These do not need to be included until you are using these APIs in the viewer.

**Review by goplayoutside3 — CHANGES_REQUESTED** [review]
Please fill out the missing PR template section "How to Review" with steps you used to test out this PR. I reviewed only with a code read so far.

File: packages/lib-subject-viewers/README.md (hunk 1 above)
Comment (kieftrav) [reply packages/lib-subject-viewers/README.md]: Done!

File: packages/lib-subject-viewers/package.json (hunk 3 above)
Comment (kieftrav) [reply packages/lib-subject-viewers/package.json:63]: Done!

File: packages/lib-subject-viewers/.gitignore (hunk 5 above)
Comment (kieftrav) [reply packages/lib-subject-viewers/.gitignore:5]: I tested all the yarn scripts and `yarn build-storybook` generates the static assets in this directory.. Which I'm assuming we don't want to commit to the repository. As a safety mechanism, I added this line so that the files that get generated from that script can be preemptively ignored.

File: packages/lib-subject-viewers/.storybook/main.js (hunk 7 above)
Comment (kieftrav) [reply packages/lib-subject-viewers/.storybook/main.js:6]: Removed

File: packages/lib-subject-viewers/preview.js (hunk 9 above)
Comment (kieftrav) [reply packages/lib-subject-viewers/preview.js]: Yup - good catch. Removed.

File: packages/lib-subject-viewers/src/translations/i18n.js (hunk 10 above)
Comment (kieftrav) [reply packages/lib-subject-viewers/src/translations/i18n.js]: Done

File: packages/lib-subject-viewers/test/setup.js (hunk 11 above)
Comment (kieftrav) [reply packages/lib-subject-viewers/test/setup.js]: Removed

**Comment by kieftrav** [comment]
@goplayoutside3 - I tightened up the code to only include what's relevant for the MVP of the `lib-subject-viewers` package. I also updated the "How to Review" section with the specific test/process I've used to make sure it works as expected. If there is anything more you'd like to see I'd appreciate additional clarification!

File: packages/lib-subject-viewers/package.json [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+		"eslint": "^8",
+    "eslint-config-next": "14.1.4",
+    "html-webpack-plugin": "~5.6.0",
+    "mocha": "~10.2.0",
+    "sinon": "~17.0.0",
+    "sinon-chai": "~3.7.0",
+    "storybook": "~7.6.11",
+    "webpack": "~5.90.0",
+    "webpack-cli": "~5.1.0",
+    "webpack-dev-server": "~5.0.3"
```
Comment (goplayoutside3) [inline packages/lib-subject-viewers/package.json:62]: As mentioned above, do you really need these in this library if it's not designed to run a local server?

**Review by goplayoutside3 — COMMENTED** [review]
Looks much better thank you! There's one last step before okay to merge -- Because we're working in a monorepo, if a third party library such as `mocha` is used in multiple libraries in the monorepo, then each package.json should have same `semvar` version. I can tell lib-subject-viewers is importing different versions of some third party packages by checking out this branch locally and running `yarn` at the root front-end-monorepo directory. Unexpected changes are made to `yarn.lock` with that command and will have a cascading effect if merged as-is (dependabot will probably go crazy trying to sync up package versions throughout the monorepo).

1) The `eslint-` packages are configured differently in this new library than others and I'm not sure why. Take a look at lib-user and lib-content for examples of `eslint-` packages and their version numbers. The parent package `eslint` is not in other package.json files in FEM.
2) Bump `mocha` in lib-subject-viewers to match the other package.json files.
3) Do you need `webpack` `webpack-cli` and `webpack-dev-server` in this library? Those look copy-pasted from lib-user, but lib-subject-viewers is not built to run a local webpack server, just a component preview with Storybook.

File: packages/lib-subject-viewers/package.json (hunk 12 above)
Comment (kieftrav) [reply packages/lib-subject-viewers/package.json]: Done!

**Comment by kieftrav** [comment]
> Looks much better thank you! There's one last step before okay to merge -- Because we're working in a monorepo, if a third party library such as `mocha` is used in multiple libraries in the monorepo, then each package.json should have same `semvar` version. I can tell lib-subject-viewers is importing different versions of some third party packages by checking out this branch locally and running `yarn` at the root front-end-monorepo directory. Unexpected changes are made to `yarn.lock` with that command and will have a cascading effect if merged as-is (dependabot will probably go crazy trying to sync up package versions throughout the monorepo).
> 
> 1. The `eslint-` packages are configured differently in this new library than others and I'm not sure why. Take a look at lib-user and lib-content for examples of `eslint-` packages and their version numbers. The parent package `eslint` is not in other package.json files in FEM.
> 2. Bump `mocha` in lib-subject-viewers to match the other package.json files.
> 3. Do you need `webpack` `webpack-cli` and `webpack-dev-server` in this library? Those look copy-pasted from lib-user, but lib-subject-viewers is not built to run a local webpack server, just a component preview with Storybook.

Next revision completed. I went through the packages and was able to do the following:
1. Remove anything related to `webpack` and `eslint`
2. Moved the current dependencies into the `peerDependencies` section because they exist as `peerDependencies` in `lib-user`
3. Made sure any package used in`lib-subject-viewers` and is used elsewhere shares the same `semver`.

**Review by goplayoutside3 — APPROVED** [review]
LGTM!
